util/jsonrpc-types/src/blockchain.rs

```diff
@@ -515,67 +516,38 @@ impl From<Transaction> for packed::Transaction {
 }
 
 /// The JSON view of a transaction as well as its status.
-#[derive(Clone, Serialize, Deserialize, PartialEq, Eq, Hash, Debug)]
-pub struct TransactionWithStatus {
+#[derive(Serialize, Deserialize, PartialEq, Eq, Hash, Debug)]
+pub struct TransactionWithStatusResponse {
     /// The transaction.
-    pub transaction: Option<TransactionView>,
+    pub transaction: Option<ResponseFormat<TransactionView>>,
     /// The Transaction status.
     pub tx_status: TxStatus,
 }
 
-impl TransactionWithStatus {
-    /// Build with pending status
-    pub fn with_pending(tx: Option<core::TransactionView>) -> Self {
-        Self {
-            tx_status: TxStatus::pending(),
-            transaction: tx.map(Into::into),
-        }
-    }
-
-    /// Build with proposed status
-    pub fn with_proposed(tx: Option<core::TransactionView>) -> Self {
-        Self {
-            tx_status: TxStatus::proposed(),
-            transaction: tx.map(Into::into),
+impl TransactionWithStatusResponse {
+    /// Transpose `tx_pool::TransactionWithStatus` to `TransactionWithStatusResponse`
+    /// according to the type of inner_type
+    pub fn from(t: tx_pool::TransactionWithStatus, inner_type: ResponseFormatInnerType) -> Self {
+        match inner_type {
+            ResponseFormatInnerType::Hex => TransactionWithStatusResponse {
+                transaction: t.transaction.map(|tx| {
+                    let packed_tx_view = packed::TransactionView::new_builder()
+                        .data(tx.data())
+                        .hash(tx.hash())
+                        .witness_hash(tx.witness_hash())
+                        .build();
+                    ResponseFormat::hex(packed_tx_view.as_bytes())
+                }),
```

Comment:
```suggestion
                transaction: t
                    .transaction
                    .map(|tx| ResponseFormat::hex(tx.data().as_bytes())),
```

Reply:
> 
OK, I did the suggestion change: 
https://github.com/nervosnetwork/ckb/compare/183f773f1ab52efdd0727bbb403538730331d093..617a21f8e88fb8f42c7a15ce15c171150935ef3a

Now, when `verbosity=0`, It will return hex encoded `packed::Transaction` instead of `packed::TransactionView`